clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/providers/TitusInstanceProvider.groovy

```diff
@@ -63,25 +68,50 @@ class TitusInstanceProvider implements InstanceProvider<TitusInstance> {
       return null
     }
 
+    if (account == null || account == "") {
+      return null
+    }
+
     String stack = awsLookupUtil.stack(account)
     if (!stack) {
       stack = 'mainvpc'
     }
 
-    CacheData instanceEntry = cacheView.get(INSTANCES.ns, Keys.getInstanceKey(id, awsAccount, stack, region))
+    Keys.CachingSchema cachingSchema = awsLookupUtil.getCachingSchemaForAccount(account)
+
+    String instanceKey = ( cachingSchema == Keys.CachingSchema.V1
+      ? Keys.getInstanceKey(id, awsAccount, stack, region)
+      : Keys.getInstanceV2Key(id, account, region))
+
+    CacheData instanceEntry = cacheView.get(INSTANCES.ns, instanceKey)
     if (!instanceEntry) {
       return null
     }
-    Job.TaskSummary task = objectMapper.convertValue(instanceEntry.attributes.task, Job.TaskSummary)
-    Job job = objectMapper.convertValue(instanceEntry.attributes.job, Job)
+    Task task = objectMapper.convertValue(instanceEntry.attributes.task, Task)
+    Job job
+    if (instanceEntry.attributes.job == null) {
+      // Instance is cached separately from job, so we must also load the job cache entry
+      if (instanceEntry.relationships[SERVER_GROUPS.ns] && !instanceEntry.relationships[SERVER_GROUPS.ns].empty) {
+        job = loadJob(instanceEntry)
+      } else {
+        log.error( "Task {} in {}:{} does not have a job", task.id, account, region)
+      }
+    } else {
+      // Instance is cached at the same time as job, V1 schema
+      objectMapper.convertValue(instanceEntry.attributes.job, Job)
+    }
+
     TitusInstance instance = new TitusInstance(job, task)
```

Comment:
Looks like we can get an NPE here with the new codepath. `@L91`, we do `Job job`, but only in a single branch in the following block of code do we assign it. Is this expected? I presume not, as at L101, you do a `objectMapper.convertValue` but do not assign its return value to anything.

Reply:
Oops good catch. L101 should assign it. Fixing.